### PR TITLE
#159987628 Users should be able to view and edit their profiles

### PR DESCRIPTION
**Pull request by iamuchejude**
#### What does this PR do?
Add view and update user profile functionality to the app

#### Description of Task to be completed?
- add the ability for a user to view his/her profile
- add the ability for users to edit their profiles
- include relevant tests

#### How should this be manually tested?
-- Pull the latest change from this repo.
-- Run `npm run dev`.
-- Navigate to `Profile Page`.
NB: Click on edit profile to update user profile

#### What are the relevant pivotal tracker stories?
[#159987628](https://www.pivotaltracker.com/story/show/159987628)

### CODE REVIEW

File: src/actions/userFollow.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import {
+  FETCH_USER_FOLLOW_SUCCESS,
+  FETCH_USER_FOLLOW_REQUEST,
+  FETCH_USER_FOLLOW_FAILED,
+} from '../actionTypes/userFollow';
+
+const API = process.env.REACT_APP_API;
+const { token } = window.localStorage;
+
+const fetchUserFollowSuccess = stat => ({
```
Comment (CEOehis) [inline src/actions/userFollow.js]: state?

File: src/components/UpdateProfile/UserPassword.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      rNewPassword: '',
+    };
+  }
+
+  onChange = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  onSubmit = e => {
+    e.preventDefault();
```
Comment (CEOehis) [inline src/components/UpdateProfile/UserPassword.js:19]: Did you send the update password request here?

Comment (iamuchejude) [reply src/components/UpdateProfile/UserPassword.js:19]: No. I initially was supposed to work on the update password before the plan changed. At that moment, I had already started implementing the feature but I had to stop thereafter.

File: src/components/UserBio/UserBio.js [hunk 3]
```diff
@@ -1,37 +1,67 @@
...
+        <span className={styles.userNames}>{`${firstName} ${lastName}`}</span>
+        <div className={styles.bioDiv}>
+          <div className={styles.userBio}>
+            {bio}
           </div>
-          <div>
-            <a href="https://linkedin.com"> <i className="fa fa-linkedin faLinkediN" /> </a>
+          {!linkedin && !twitter ? '' : <div className={styles.socialMedia}>
+            <ul>
+              {!twitter ? '' : <li>
```
Comment (emekafredy) [inline src/components/UserBio/UserBio.js:40]: After updating my twitter field, clicking on the link in my profile should take me to Twitter, but it doesn't at the moment. This is what I get.
screen shot 2018-10-25 at 10 03 52 am

File: src/components/UpdateProfile/UserInfo.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+          </div>
+          <div className={styles.form_input_group}>
+            <label htmlFor="Role">Role</label>
+            <input type="text" onChange={this.onChange} readOnly name="role" value={role || ''} placeholder="Role" />
+          </div>
+          <div className={styles.form_input_group}>
+            <label htmlFor="Twitter">Twitter</label>
+            <input type="text" onChange={this.onChange} name="twitter" value={twitter || ''} placeholder="Twitter profile URL" />
+          </div>
+          <div className={styles.form_input_group}>
```
Comment (emekafredy) [inline src/components/UpdateProfile/UserInfo.js:93]: It seems I can't update my linkedin field with a valid url
screen shot 2018-10-25 at 10 03 27 am

File: src/components/UpdateProfile/UserPassword.js (hunk 2 above)
Comment (CEOehis) [reply src/components/UpdateProfile/UserPassword.js:19]: Ok. That makes sense.

**Review by emekafredy — CHANGES_REQUESTED** [review]
Nice work bro. Just a few changes to implement. Check your comment

**Review by CEOehis — APPROVED** [review]
👍LGTM

**Review by emekafredy — APPROVED** [review]
LGTM

**Review by johnkingzy — APPROVED** [review]
LGTM

**Pull request merged**